src/core/localize.rkt

```diff
@@ -154,20 +180,80 @@
                (vector-ref exacts (vector-member idx roots)))) ; arg's index mapping to exact
            (define approx (apply (impl-info f 'fl) argapprox))
            (ulp-difference exact approx repr)]))
-      (vector-set! (vector-ref errs expr-idx) pt-idx err)))
+      (vector-set! (vector-ref exacts-out expr-idx) pt-idx exact)
+      (vector-set! (vector-ref errs expr-idx) pt-idx err)
+      (vector-set! (vector-ref actuals-out expr-idx) pt-idx actual)))
 
   (define n 0)
   (for/list ([subexprs (in-list subexprss)])
     (for*/hash ([subexpr (in-list subexprs)])
-      (begin0 (values subexpr (vector->list (vector-ref errs n)))
+      (begin0 (values subexpr
+                      (hasheq 'errs
+                              (vector->list (vector-ref errs n))
+                              'exact-values
+                              (vector->list (vector-ref exacts-out n))
+                              'actual-values
+                              (vector->list (vector-ref actuals-out n))))
         (set! n (add1 n))))))
 
 ;; Compute the local error of every subexpression of `prog`
 ;; and returns the error information as an S-expr in the
 ;; same shape as `prog`
-(define (local-error-as-tree expr ctx)
-  (define errs (first (compute-local-errors (list (all-subexpressions expr)) ctx)))
-  (let loop ([expr expr])
-    (match expr
-      [(list op args ...) (cons (hash-ref errs expr) (map loop args))]
-      [_ (list (hash-ref errs expr))])))
+(define (local-error-as-tree test ctx)
```

Comment:
Kind of inelegant but I'm willing to merge this, I believe only Odyssey uses this code path.